.github/workflows/backport-command.yml

```diff
@@ -57,11 +93,171 @@ jobs:
+          FIXING_ISSUE_URLs: ${{ steps.pr_details.outputs.fixing_issue_urls }}
+          ORIG_PR_NUMBER: ${{ github.event.client_payload.pull_request.number }}
+          ORIG_PR_URL: ${{ github.event.client_payload.pull_request.html_url }}
+        run: |
+          backport_issue_urls=""
+          if [[ "$FIXING_ISSUE_URLs" != "" ]]; then
+            for FIXING_ISSUE_URL in $FIXING_ISSUE_URLs; do
+              orig_issue_number=$(echo "$FIXING_ISSUE_URL" | awk -F/ '{print $NF}')
+              orig_org=$(echo "$FIXING_ISSUE_URL" | awk -F/ '{print $4}')
+              orig_repo=$(echo "$FIXING_ISSUE_URL" | awk -F/ '{print $5}')
+
+              if [[ "$orig_repo" != "${TARGET_REPO}" || "$orig_org" != "${TARGET_ORG}" ]]; then
+                break
+              fi
+
+              orig_issue_title=$(gh issue view "$orig_issue_number" \
+                --repo "${orig_org}/${orig_repo}" \
+                --json title --jq .title)
+
+              backport_issue_url=$(gh issue list --repo "${orig_org}/${orig_repo}" \
+                --state open \
+                --search "[${BACKPORT_BRANCH}] ${orig_issue_title} in:title" \
+                --json url \
+                --milestone "${TARGET_MILESTONE}" \
+                --jq '.[0].url')
+
+              if [[ "$backport_issue_url" == "" ]]; then
+                # backport issue does not exist and will be created
+                # get orig issue assignees
+                orig_issue_assingees=$(gh issue view "$orig_issue_number" \
+                  --repo "${orig_org}/${orig_repo}" \
+                  --json assignees --jq .assignees.[].login | paste -s -d ',' -)
+                # create issue
+                backport_issue_url=$(gh issue create --title "[${BACKPORT_BRANCH}] ${orig_issue_title}" \
+                  --label "kind/backport" \
+                  --repo "${orig_org}/${orig_repo}" \
+                  --assignee "${orig_issue_assingees}" \
+                  --milestone "${TARGET_MILESTONE}" \
+                  --body "Backport ${FIXING_ISSUE_URL} to branch ${BACKPORT_BRANCH}. Requested by PR $ORIG_PR_URL")
+              fi
+              backport_issue_urls+=$(echo "Fixes $backport_issue_url, ")
+            done
+            backport_issue_urls=$(echo $backport_issue_urls | sed 's/.$//')
+          fi
+
+          gh pr create --title "[${BACKPORT_BRANCH}] ${ORIG_TITLE}" \
+            --base "${BACKPORT_BRANCH}" \
+            --label "kind/backport" \
+            --head "vbotbuildovich:${HEAD_BRANCH}" \
+            --draft \
             --repo "${TARGET_ORG}/${TARGET_REPO}" \
-            --assignee "${ORIG_ASSIGNEES}" \
+            --reviewer "${ORIG_REVIEWERS}" \
             --milestone "${TARGET_MILESTONE}" \
-            --body "Backport ${ORIG_ISSUE_URL} to branch ${BACKPORT_BRANCH}"
+            --body """Backport from pull request: ${ORIG_ISSUE_URL}
+            $backport_issue_urls
+            """
```

Comment:
would https://github.com/marketplace/actions/create-pull-request work here? the action also has `branch-suffix` strategies to help keep the branch name unique.

Reply:
that's interesting, but I see that they don't support cherry-picking commits so we wouldn't avoid some manual `git` steps. If you strongly prefer it then I can try it

Reply:
@andrewhsu although this makes me think... should we produce a completely random branch name?? if so, we'll avoid some special characters that are not accepted as branch name (eg white spaces). currently it replaces all whitespaces with '-' but maybe it's better to get a hash of the string for example as branch name. it'll be confusing though cause they won't be descriptive... wdyt??

Reply:
resolving this because code has changed. i'll put suggestion in updated code.